src/CloudSearchDomain/CloudSearchDomainFactory.php

```diff
@@ -26,7 +26,7 @@ protected function addDefaultArgs(&$args)
         if (strpos($args['endpoint'], 'http') !== 0) {
             $args['endpoint'] = Url::buildUrl([
                 'host'   => $args['endpoint'],
-                'scheme' => $args['scheme'],
+                'scheme' => 'https',
```

Comment:
What if we just updated the getDomainClient() function to return an absolute URL? That could do the user a favor of assuming a default scheme of https, while not introducing a one-off behavior of a single client (i.e., endpoint accepting a domain and not an actual endpoint).